packages/gatsby/cache-dir/static-query.js

```diff
@@ -2,9 +2,42 @@ import React from "react"
 import PropTypes from "prop-types"
 
 const StaticQueryContext = React.createContext({})
-let StaticQueryServerContext = null
+let _StaticQueryServerContext = null
 if (React.createServerContext) {
-  StaticQueryServerContext = React.createServerContext(`StaticQuery`, {})
+  _StaticQueryServerContext = createServerContext(`StaticQuery`, {})
+}
+
+function createServerOrClientContext(name, defaultValue) {
```

Comment:
Would be a good idea to move these generic functions to their own module and import here, it's likely we'll use them more than once